### PR TITLE
Add reboot notification to dsc_resource

### LINKED ISSUE #4156: dsc_resource doesn't have a return property
We are using the chef dsc_resource to invoke an experimental dsc resource xSharePoint, and when running the pre-req installer it returns an error code which means we need to reboot, we are unable to use the return property like with the chef powershell_script. the error we get is: **undefined method 'returns' for chef::resource::dscresource.**

## Example with powershell_script:
```
powershell_script "install-#{cab}" do
    code <<-EOH
      DISM.exe /Online /Add-Package /PackagePath:#{node['stores-ISP-common']['chef_temp']}\\#{cab} /quiet /norestart
       Del #{node['stores-ISP-common']['chef_temp']}\\#{cab.split('.')[0]}.*
     EOH
     returns [0, 3010]
     notifies :reboot_now, "reboot[reboot-computer]", :immediately if returns 3010
     returns 0
     not_if "wmic qfe | findstr /i '#{cab.split('-')[2]}'"
   end
```
## **Failed Example with dsc_resource:**
```
dsc_resource 'Install_SP' do
   resource :xSPInstallPrereqs
   property :path, "C:\Users\Public\Documents\example.zip"
   returns [0, 30066]
   notifies :reboot_now, "reboot[reboot-computer]", :immediately if returns 30066
   returns 0
 end
```
I wanted to open an issue to see if the returns property should be general functionality to all chef resources, or if we should find a different way to handle our reboots.

### PR DESCRIPTION
**Pull request by smurawski**
Adds an allow_reboot attribute to dsc_resource.

Example:
```
dsc_resource 'Install AD' do
  resource :windowsfeature
  property :name, 'ad-domain-services'
  allow_reboot :reboot_now
end
```

Closes #4156

### CODE REVIEW

**Comment by smurawski** [comment]
Thoughts @chef/client-windows ?

File: DOC_CHANGES.md [hunk 1]
```diff
@@ -56,3 +56,10 @@ Assuming both of those make 12.6, placeholder pages:
 chocolatey_package: https://docs.chef.io/release/12-6/resource_chocolatey_package.html
 
 ksh: https://docs.chef.io/release/12-6/resource_ksh.html
+
+### `dsc_resource` resource
+
+Added allow_reboot attribute to dsc_resource.
```
Comment (mwrock) [inline DOC_CHANGES.md]: We should explicitly state that the default is `:nothing`

**Comment by mwrock** [comment]
:+1: with my nitpick above

**Comment by smurawski** [comment]
closing this in favor of #4167

**Pull request closed without merging**